homeassistant/components/dwd_weather_warnings/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow for the dwd_weather_warnings integration."""
+
+from __future__ import annotations
+
+from typing import Any, Final
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_MONITORED_CONDITIONS, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+
+from .const import (
+    ADVANCE_WARNING_SENSOR,
+    CONF_OLD_REGION_NAME,
+    CONF_REGION_IDENTIFIER,
+    CURRENT_WARNING_SENSOR,
+    DEFAULT_MONITORED_CONDITIONS,
+    DEFAULT_NAME,
+    DOMAIN,
+    LOGGER,
+)
+
+CONFIG_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_REGION_IDENTIFIER): cv.string,
+        vol.Required(
+            CONF_MONITORED_CONDITIONS, default=DEFAULT_MONITORED_CONDITIONS
+        ): SelectSelector(
+            SelectSelectorConfig(
+                options=[
+                    SelectOptionDict(
+                        label="Current warning level",
+                        value=CURRENT_WARNING_SENSOR,
+                    ),
+                    SelectOptionDict(
+                        label="Advance warning level",
+                        value=ADVANCE_WARNING_SENSOR,
+                    ),
+                ],
+                multiple=True,
+            )
+        ),
+    }
+)
+
+
+class DwdWeatherWarningsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for the dwd_weather_warnings integration."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict = {}
+
+        if user_input is not None:
+            # Check, if either CONF_REGION or CONF_GPS_TRACKER has been set.
```

Comment:
Not needed as choosing a region is mandatory in the schema

Reply:
True, forgot to change that part from my old PR where the new feature was already implemented